src/main.rs

```diff
@@ -355,3 +383,45 @@ fn error_404(req: &HttpRequest<crate::MiniserveConfig>) -> Result<HttpResponse,
         .into_string(),
     ))
 }
+
+// Print the QR code of the specified content in the terminal
+// For the width and height to be equal, two characters are used in each grid
+// Cannot control the line-height of the terminal, so it looks imperfect
+fn print_qrcode(content: String) {
+    let qr = QrCode::encode_text(&content, QrCodeEcc::Medium).unwrap();
+    let size = qr.size();
+    let margin = 1;
+    
+    let mut text = String::new();
+    let white = Paint::white("██").to_string();
+    let black = Paint::black("  ").to_string();
+
+    for x in -margin..size + margin {
+        for y in -margin..size + margin {
+            // margin
+            if x < 0 || x > size || y < 0 || y > size {
+                text.push_str(&white);
+                continue;
+            }
+            // QR code fill
+            if qr.get_module(x, y) {
+                text.push_str(&black);
+            } else {
+                text.push_str(&white);
+            }
+        }
+        text.push_str("\n");
+    }
+
+    print!("{}", text);
+}
+
+fn get_intranet_ip() -> Option<IpAddr> {
+    pnet_datalink::interfaces()
```

Comment:
Why exactly did you pull in this dependency?

Reply:
Get local IP address.

Reply:
I'm concerned about this apparently requring winpcap on Windows which think we can't link statically. I'd very much like this to stay a single static binary on Windows. I just checked this with `cargo-cross` but you could also check this natively.

Reply:
Myabe the crate [systemstat](https://github.com/myfreeweb/systemstat) will be an option. it support get the interfaces on diffferent os (contion windwos also).